### ISSUE TITLE
Every file in the repo shown as "new" after switching branch

### ISSUE THREAD

**Issue by tallakt** [desktop/desktop#18068]
### The problem

I am working with a repository on Mac OS. Now and then, all files in the repository are shown as "green" in the file browser, or as new files. I can do a "discard all changes" but the system will quickly revert to all files "green". Doing that would also remove all my useful changes.

I found another method; I open the terminal and run

```
git add .
```

After doing that, everything looks fine. Only changed files are shown in the file pane of GitHub Desktop.

This is very frustrating and really makes GitHub Desktop unusable. I have used it for a long time, but this issue only appeared during the last few months. When it happens, it will be all the time. Other times, no issues.

Thanks.

### Release version

3.3.7 (Arm64)

### Operating system

Darwin pi.local 23.1.0 Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000 arm64

### Steps to reproduce the behavior

Check out my branch `dev_tt_verify_tether_sim`

### Screenshots

Screen shot showing the files pane after switching to a branch, without any local changes done

![image](https://github.com/desktop/desktop/assets/49966/ed942e21-a5f2-4f53-98e8-e2726b3b9b72)

**Comment by steveward** [comment]
@tallakt thanks for the issue. Usually this is caused by file mode or line endings changes. Can you run `git diff` against one of the files to see if its showing a file mode change?

**Comment by tallakt** [comment]
Nothing in the diff, literally

```
sw (dev_tt_verify_tether_sim +) % git diff -- changelog.markdown
sw (dev_tt_verify_tether_sim +) % git diff
```

If I diff towards `HEAD` I get:

```
sw (dev_tt_verify_tether_sim +) % git diff HEAD -- changelog.markdown
```

```
diff --git a/changelog.markdown b/changelog.markdown
deleted file mode 100644
index d1d9d7ca..00000000
--- a/changelog.markdown
+++ /dev/null
@@ -1,234 +0,0 @@
-## Software changes
-
-### Testweek 42 2023

```

Just to be clear, the files are not deleted either on the file system or in HEAD

**Comment by tallakt** [comment]
Hi. I am adding this incident to the case because I think it may be related. 

Today I lost many hours of work after switching branches in Github Desktop on Mac. 

I had file changes that were not checked in, then switched branch. I selected "leave changes on...". Still, when changing back to the original branch, there was no stash to bring back, not in the desktop client nor in the command line. I know I did not "bring the changes to the new branch" because that operation would also not have deleted my changed files.

In many years of using git, this is the first time I have lost any file changes due to a bug. It is serious enough that I am considering not using Github Desktop anymore.

**Comment by steveward** [comment]
@tallakt can you send over the log file from the day this happened? It's possible that we can recover those changes, similar to https://github.com/desktop/desktop/issues/11622.

**Comment by tallakt** [comment]
I was able to recover most of my work as the deleted file was still in memory of my editor. so we don't need to recover the files.

The incident still was worrying, because this is the first time "git" accidentally deleted my work.

What happened to me though looks similar to https://github.com/desktop/desktop/issues/11622 


This was in my log, though I did not find any relevant info. I am unsure of the exact time that the files were disappeared.

```
2024-02-12T09:19:31.159Z - warn: [ui] Attempted to add a popup of already existing type - StashAndSwitchBranch.
2024-02-12T09:19:31.160Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.000s
2024-02-12T09:19:38.243Z - warn: [ui] Attempted to add a popup of already existing type - StashAndSwitchBranch.
2024-02-12T09:19:38.243Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.000s
2024-02-12T09:19:45.192Z - warn: [ui] Attempted to add a popup of already existing type - StashAndSwitchBranch.
2024-02-12T09:19:45.194Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.000s
2024-02-12T09:19:58.156Z - info: [ui] [AppStore] loading 28 repositories from store
2024-02-12T09:19:58.169Z - info: [ui] [AppStore] found account: tallakt (Tallak Tveide)
2024-02-12T09:19:58.182Z - info: [ui] [BranchPruner] Last prune took place 3 hours ago - skipping
2024-02-12T09:19:58.568Z - info: [ui] Subscribed 'tallakt' to Alive channel
2024-02-12T09:19:58.610Z - info: [ui] launching: 3.3.8 (Mac OS 14.1.2)
2024-02-12T09:19:58.610Z - info: [ui] execPath: '/Applications/GitHub Desktop.app/Contents/Frameworks/GitHub Desktop Helper (Renderer).app/Contents/MacOS/GitHub Desktop Helper (Renderer)'
2024-02-12T09:20:02.694Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.001s
2024-02-12T09:20:05.310Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T09:20:05.666Z - info: [ui] [Timing] Action 'stash and checkout' for 'kitemill/KiteSim' took 0.354s
2024-02-12T10:15:23.604Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T10:15:23.686Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.076s
2024-02-12T10:15:27.472Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: bumblebee - tallakt (has token)
2024-02-12T10:15:32.852Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 27 repositories took 12.9s of which 0.0s paused, total 12.9s
2024-02-12T10:15:35.143Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T10:15:35.220Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.073s
2024-02-12T10:15:37.195Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T10:15:37.264Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.068s
2024-02-12T10:52:52.737Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 27 repositories took 10.2s of which 0.0s paused, total 10.2s
2024-02-12T11:00:16.018Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T11:00:16.117Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.090s
2024-02-12T11:01:18.919Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.001s
2024-02-12T11:01:21.107Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T11:01:21.204Z - info: [ui] [Timing] Action 'stash and checkout' for 'kitemill/KiteSim' took 0.095s
2024-02-12T11:01:26.022Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.000s
2024-02-12T11:01:30.240Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T11:01:30.357Z - info: [ui] [Timing] Action 'stash and checkout' for 'kitemill/KiteSim' took 0.116s
2024-02-12T11:01:33.204Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T11:01:33.261Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.054s
2024-02-12T11:01:37.332Z - info: [ui] [AppStore. _popStashEntry] popped stash with commit id ab7e67b2757cb8e35035e174682d926cd8fb5075
2024-02-12T12:04:50.882Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.000s
2024-02-12T12:04:54.196Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T12:04:54.307Z - info: [ui] [Timing] Action 'stash and checkout' for 'kitemill/KiteSim' took 0.105s
2024-02-12T12:04:55.233Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 27 repositories took 10.3s of which 0.0s paused, total 10.3s
2024-02-12T12:04:56.633Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T12:04:56.729Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.093s
2024-02-12T12:04:59.981Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T12:05:00.070Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.078s
2024-02-12T13:19:58.172Z - info: [ui] [BranchPruner] Last prune took place 7 hours ago - skipping
2024-02-12T16:04:57.538Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 27 repositories took 11.8s of which 0.0s paused, total 11.8s
2024-02-12T16:09:46.502Z - info: [ui] [Timing] Action 'create commit' for 'kitemill/KiteSim' took 0.104s
2024-02-12T16:09:47.817Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T16:09:48.936Z - info: [ui] Executing push: git -c credential.helper= push origin main:main --progress (took 1.115s)
2024-02-12T16:09:50.994Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-12T16:09:51.054Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.058s
2024-02-12T16:41:27.204Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 27 repositories took 10.4s of which 0.0s paused, total 10.4s
2024-02-12T17:19:58.282Z - info: [ui] [BranchPruner] Last prune took place 11 hours ago - skipping
2024-02-12T21:19:58.233Z - info: [ui] [BranchPruner] Last prune took place 15 hours ago - skipping

```

**Comment by tallakt** [comment]
Yesterday I opted to stash in git manually in the terminal. After doing that I changed branch to `main` then back today. My stash was gone.

I have never seen anything like this before.

[the top terminal is today, trying to recover the stash, the bottom terminal has been untouched since yesterday when I ran the git stash command]

![image](https://github.com/desktop/desktop/assets/49966/191009e5-ec06-4a19-a3f5-eac0e3760af5)


The log file for yesterday

```
2024-02-13T01:19:58.123Z - info: [ui] [BranchPruner] Last prune took place 19 hours ago - skipping
2024-02-13T05:19:58.012Z - info: [ui] [BranchPruner] Last prune took place 23 hours ago - skipping
2024-02-13T07:29:27.888Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: bumblebee - tallakt (has token)
2024-02-13T07:29:29.046Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 1.148s)
2024-02-13T08:58:20.938Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: BOSL2 - tallakt (has token)
2024-02-13T08:58:25.968Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 27 repositories took 13.8s of which 5331.1s paused, total 5344.9s
2024-02-13T09:03:49.103Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 27 repositories took 10.2s of which 0.0s paused, total 10.2s
2024-02-13T09:19:57.790Z - info: [ui] [BranchPruner] No branches to prune.
2024-02-13T09:19:58.583Z - info: [ui] Stats reported.
2024-02-13T12:33:33.677Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: bumblebee - tallakt (has token)
2024-02-13T12:33:39.392Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 27 repositories took 12.0s of which 0.0s paused, total 12.0s
2024-02-13T13:05:20.886Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 27 repositories took 11.1s of which 19.1s paused, total 30.2s
2024-02-13T13:05:22.446Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-13T13:05:22.531Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.083s
2024-02-13T13:19:57.718Z - info: [ui] [BranchPruner] Last prune took place 4 hours ago - skipping
2024-02-13T17:19:57.855Z - info: [ui] [BranchPruner] Last prune took place 8 hours ago - skipping
2024-02-13T17:21:18.860Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 1.002s)
2024-02-13T21:19:57.774Z - info: [ui] [BranchPruner] Last prune took place 12 hours ago - skipping
```

And for today

```
2024-02-14T01:19:57.701Z - info: [ui] [BranchPruner] Last prune took place 16 hours ago - skipping
2024-02-14T05:19:57.610Z - info: [ui] [BranchPruner] Last prune took place 20 hours ago - skipping
2024-02-14T06:12:30.585Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 1.015s)
2024-02-14T07:22:15.620Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: bumblebee - tallakt (has token)
2024-02-14T07:22:17.757Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: bldc - tallakt (has token)
2024-02-14T07:22:21.989Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: MegAWES - tallakt (has token)
2024-02-14T07:22:23.558Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 27 repositories took 15.5s of which 0.0s paused, total 15.5s
2024-02-14T07:22:25.739Z - info: [ui] [Timing] Action 'create commit' for 'kitemill/KiteSim' took 0.090s
2024-02-14T07:22:28.099Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-14T07:22:30.035Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: KiteSim - tallakt (has token)
2024-02-14T07:22:30.119Z - info: [ui] [Timing] Action 'checkout branch from list' for 'kitemill/KiteSim' took 0.083s
2024-02-14T07:27:14.429Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 1.721s)
2024-02-14T07:27:16.061Z - info: [ui] Executing updateRemoteHEAD: git -c credential.helper= remote set-head -a origin (took 1.633s)
```

**Comment by tallakt** [comment]
I did some more investigating on this. I copied my repo folder from `KiteSim` to `KiteSim_` then cloned the repository new. Using `diff -qr` I compared the two folders and found a few things different:

```
GitHub  % diff KiteSim/.git/config KiteSim_/.git/config 
15a16,17
> [lfs]
> 	repositoryformatversion = 0
```

Next I ran a command called `fs_usage` and found a line

```
10:29:47  rmdir             /Users/tallakt/Documents/GitHub/sw/.gitattributes                                0.000004   fileprovider
```

I was able to guess that maybe iCloud sync was to blame after searching for info on `fileprovider`. I moved the whole `GitHub` folder out of the Documents folder and it appears that my problems are gone for now.

I would maybe suggest the default folder location for GitHub Desktop be placed in the home folder rather than the Documents folder, to get away from this issue. I think it makes less sense to sync git repositories anyway, both as they are already backed up through git, and also because i bypasses any security set up in git.
